backend/migrations/20230310_000000_migration.sql

```diff
@@ -0,0 +1,283 @@
+, account_id UUID REFERENCES accounts_v0(id) NOT NULL
+, name VARCHAR(64) UNIQUE NOT NULL
+, created_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, updated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+CREATE TRIGGER set_canvas_timestamp
+BEFORE UPDATE ON canvases_v0
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_set_timestamp();
+
+
+CREATE TABLE IF NOT EXISTS
+cron_records_v0
+( id SERIAL PRIMARY KEY
+, tlid BIGINT NOT NULL
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, ran_at TIMESTAMPTZ NOT NULL DEFAULT NOW ()
+);
+
+CREATE INDEX IF NOT EXISTS
+idx_cron_records_tlid_canvas_id_id
+ON cron_records_v0
+(tlid, canvas_id, id DESC);
+
+
+CREATE TABLE IF NOT EXISTS
+custom_domains_v0
+( host TEXT PRIMARY KEY
+, canvas TEXT);
+
+
+CREATE TABLE IF NOT EXISTS
+events_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID NOT NULL
+, module TEXT NOT NULL
+, name TEXT NOT NULL
+, modifier TEXT NOT NULL
+, locked_at TIMESTAMPTZ -- nullable
+, enqueued_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
+, value TEXT NOT NULL
+);
+
+--#[no_tx]
+
+-- We want to use this index to:
+-- 1) count the number of items in this queue, so it's important that the entire
+-- search term is in the index or it will need to hit disk. This is true even though
+-- the module rarely changes
+-- 2) fetch the indexes for all items we're unpausing. This is rare so it's fine to
+CREATE INDEX CONCURENTLY IF NOT EXISTS
+idx_events_count
+ON events_v0 (canvas_id, module, name);
+
+
+CREATE TABLE IF NOT EXISTS
+function_arguments_v0
+( id UUID PRIMARY KEY
+, canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, tlid BIGINT NOT NULL
+, trace_id UUID; -- nullable for now
+, timestamp TIMESTAMPTZ NOT NULL
+, arguments_json TEXT NOT NULL
+);
+
+CREATE INDEX IF NOT EXISTS
+function_arguments_most_recent
+ON function_arguments_v0
+(canvas_id, tlid, timestamp DESC);
+
+CREATE INDEX IF NOT EXISTS
+function_arguments_for_trace
+ON function_arguments_v0
+(canvas_id, tlid, trace_id);
+
+
+CREATE TABLE IF NOT EXISTS
+function_results_v0
+( canvas_id UUID REFERENCES canvases_v0(id) NOT NULL
+, tlid BIGINT NOT NULL
+, fnname TEXT NOT NULL
+, id BIGINT NOT NULL
+, hash TEXT NOT NULL
+, timestamp TIMESTAMPTZ NOT NULL
+, value TEXT NOT NULL
+, trace_id UUID NOT NULL
+, hash_version INTEGER
+);
+
+CREATE INDEX IF NOT EXISTS
+idx_function_results_v0_most_recent
+ON function_results_v0
+(canvas_id, trace_id, tlid, timestamp DESC);
+
+
+ /* associate it back to the function. */
+CREATE TABLE IF NOT EXISTS packages_v0
+( id UUID PRIMARY KEY
+, tlid BIGINT
+  /* owner/namespace part of the string, eg dark */
+, user_id UUID REFERENCES accounts_v0 (id) NOT NULL
+, package TEXT NOT NULL /* eg stdlib */
+, module TEXT NOT NULL /* eg Twitter */
+, fnname TEXT NOT NULL /* eg sendText */
+, version INTEGER NOT NULL /* eg 0 */
+, description TEXT NOT NULL /* docstring */
+, body BYTEA NOT NULL
+, return_type TEXT NOT NULL
+, parameters jsonb NOT NULL
+, author_id UUID REFERENCES accounts_v0(id) NOT NULL /* who uploaded this */
```

Comment:
Do `author_id` and `user_id` still need to reference `accounts_id` to identify who uploaded the package? Should we keep one, both, or neither of these references?

Reply:
They're still needed, but they shouldn't be foreign keys

Reply:
Removed FKs and kept `author_id` and `user_id` fields